pkg/pillar/types/volumetypes.go

```diff
@@ -0,0 +1,182 @@
+}
+
+// VolumeStatus is response from volumemgr about status of volumes
+type VolumeStatus struct {
+	VolumeID                uuid.UUID
+	ContentID               uuid.UUID
+	VolumeContentOriginType zconfig.VolumeContentOriginType
+	MaxVolSize              uint64
+	ReadOnly                bool
+	GenerationCounter       int64
+	DisplayName             string
+	State                   SwState
+	RefCount                uint
+	Progress                uint   // In percent i.e., 0-100
+	FileLocation            string // Location of filestystem
+	VolumeCreated           bool   // Done aka Activated
+	ContentFormat           zconfig.Format
+	LastUse                 time.Time
+	PreReboot               bool // Was volume last use prior to device reboot?
+
+	ErrorAndTimeWithSource
+}
+
+// Key is volume UUID which will be unique
+func (status VolumeStatus) Key() string {
+	return fmt.Sprintf("%s.%d", status.VolumeID.String(), status.GenerationCounter)
+}
+
+// IsContainer will return true if content tree attached
+// to the volume is of container type
+func (status VolumeStatus) IsContainer() bool {
+	if status.ContentFormat == zconfig.Format_CONTAINER {
+		return true
+	}
+	return false
+}
+
+// LogCreate :
+func (status VolumeStatus) LogCreate() {
+	logObject := base.NewLogObject(base.VolumeStatusLogType, status.DisplayName,
+		status.VolumeID, status.LogKey())
+	if logObject == nil {
+		return
+	}
+	logObject.CloneAndAddField("content-id", status.ContentID).
+		AddField("max-vol-size-int64", status.MaxVolSize).
+		AddField("state", status.State).
+		AddField("progress-int64", status.Progress).
+		AddField("refcount-int64", status.RefCount).
+		AddField("fileLocation", status.FileLocation).
+		Infof("Volume status create")
+}
+
+// LogModify :
+func (status VolumeStatus) LogModify(old interface{}) {
+	logObject := base.EnsureLogObject(base.VolumeStatusLogType, status.DisplayName,
+		status.VolumeID, status.LogKey())
+
+	oldStatus, ok := old.(VolumeStatus)
+	if !ok {
+		log.Errorf("LogModify: Old object interface passed is not of VolumeStatus type")
+	}
+	if oldStatus.ContentID != status.ContentID ||
+		oldStatus.MaxVolSize != status.MaxVolSize ||
+		oldStatus.State != status.State ||
+		oldStatus.Progress != status.Progress ||
+		oldStatus.RefCount != status.RefCount ||
+		oldStatus.FileLocation != status.FileLocation {
+
+		logObject.CloneAndAddField("content-id", status.ContentID).
+			AddField("max-vol-size-int64", status.MaxVolSize).
+			AddField("state", status.State).
+			AddField("progress-int64", status.Progress).
+			AddField("refcount-int64", status.RefCount).
+			AddField("fileLocation", status.FileLocation).
+			AddField("old-content-id", oldStatus.ContentID).
+			AddField("old-max-vol-size-int64", oldStatus.MaxVolSize).
+			AddField("old-state", oldStatus.State).
+			AddField("old-progress-int64", oldStatus.Progress).
+			AddField("old-refcount-int64", oldStatus.RefCount).
+			AddField("old-fileLocation", oldStatus.FileLocation).
```

Comment:
Ditto

Reply:
Corrected. Missed it out somehow. Sorry.